src/enforcer.js

```diff
@@ -225,25 +294,44 @@ export class TrustedTypesEnforcer {
       return;
     }
 
-    if ('ShadowRoot' in window) {
-      this.restoreSetter_(ShadowRoot.prototype, 'innerHTML');
+    if ('ShadowRoot' in this.windowObject_) {
+      this.restoreSetter_(this.windowObject_.ShadowRoot.prototype, 'innerHTML');
     }
-    this.restoreFunction_(Range.prototype, 'createContextualFragment');
-    this.restoreFunction_(insertAdjacentObject, 'insertAdjacentHTML');
-    this.restoreFunction_(Element.prototype, 'setAttribute');
-    this.restoreFunction_(Element.prototype, 'setAttributeNS');
 
-    if (getOwnPropertyDescriptor(Document.prototype, 'write')) {
-      this.restoreFunction_(Document.prototype, 'write');
-    } else {
+    if (this.windowObject_.Range) {
+      this.restoreFunction_(
+          this.windowObject_.Range.prototype,
+          'createContextualFragment'
+      );
+    }
+    this.restoreFunction_(
+        this.insertAdjacentObjectPrototype,
+        'insertAdjacentHTML'
+    );
+    this.restoreFunction_(
+        this.windowObject_.Element.prototype, 'setAttribute');
+    this.restoreFunction_(
+        this.windowObject_.Element.prototype, 'setAttributeNS');
+
+    if (getOwnPropertyDescriptor(
+        this.windowObject_.Document.prototype, 'write')
+    ) {
+      this.restoreFunction_(this.windowObject_.Document.prototype, 'write');
+    } else if (this.windowObject_.HTMLDocument && getOwnPropertyDescriptor(
+        this.windowObject_.HTMLDocument, 'write')
```

Comment:
`this.windowObject_.HTMLDocument.prototype` (this is what broke the tests)

Reply:
Yep, verified with Firefox binary 63 locally.